android-agent/src/main/java/io/opentelemetry/android/instrumentation/AndroidInstrumentation.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import android.app.Application
+import io.opentelemetry.android.OpenTelemetryRum
+
+interface AndroidInstrumentation {
```

Comment:
Just to confirm, the idea is that each `AndroidInstrumentation` works independently, and they only need to interact with `OpenTelemetryRum` to access the OTel SDK bits (e.g. Tracer)? Will the Agent ever need to reference the Instrumentations at all, for validations, throttling, or other needs?

Reply:
> Just to confirm, the idea is that each AndroidInstrumentation works independently, and they only need to interact with OpenTelemetryRum to access the OTel SDK bits (e.g. Tracer)?

I don't remember if we discussed this in the SIG, but yes, this is correct.

> Will the Agent ever need to reference the Instrumentations at all, for validations, throttling, or other needs?

Right now that's not the case, the agent doesn't need to do anything else with the instrumentations other than "activating" them by calling `AndroidInstrumentation.apply` (or `install`, depending on what name we decide to use) to provide them with the rum object plus the Application object. Though this can be changed if needed, I just closed this PR to start working on the changes in smaller PRs where we could discuss it all more in detail.